src/pyubx2/ubxreader.py

```diff
@@ -293,6 +296,19 @@ def _read_bytes(self, size: int) -> bytes:
         :rtype: bytes
         :raises: EOFError if stream ends prematurely
         """
+        if self._timeout is not None:

```

Comment:
I don't believe this change would be necessary if UPD-WRITE was defined with variable length - see other comments